lambda/update/attorney_opt_out.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import "github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
+
+type AttorneyOptOut struct {
+	AttorneyUID string
+}
+
+func (c AttorneyOptOut) Apply(lpa *shared.Lpa) []shared.FieldError {
+	attorney, ok := lpa.GetAttorney(c.AttorneyUID)
+	if !ok {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney not found"}}
+	}
+
+	if attorney.SignedAt != nil && !attorney.SignedAt.IsZero() {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney cannot opt out after signing"}}
+	}
+
+	// TODO Check if status correct
+	attorney.Status = shared.AttorneyStatusRemoved
```

Comment:
Makes sense to stick with removed for now. If we find a need for something more particular then we can add that later (either a new status, or a new "reason for removal"-type field)